FolderObserver.getExtension() fails on startup and halts openHAB

From time to time during startup I get the following error which finally leads openHAB completely not work anymore. After some reboots it works again (openHAB 5.0.0.M3):

```
"2025-07-08 07:45:17.069 [ERROR] [.core.internal.folder.FolderObserver] - bundle org.openhab.core.model.core:5.0.0.M3 (197)[org.openhab.core.model.core.internal.folder.FolderObserver(78)] : The addModelParser method has thrown an exception
java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileName()" because "path" is null
	at org.openhab.core.model.core.internal.folder.FolderObserver.getExtension(FolderObserver.java:288) ~[?:?]
	at org.openhab.core.model.core.internal.folder.FolderObserver.processIgnoredPaths(FolderObserver.java:176) ~[?:?]
	at org.openhab.core.model.core.internal.folder.FolderObserver.addModelParser(FolderObserver.java:102) ~[?:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]

```

which relates to the following code

```
    private static @Nullable String getExtension(Path path) {
        return getExtension(path.getFileName().toString());
    }

called by 

    private void processIgnoredPaths(String extension) {
        logger.debug("Processing {} ignored paths for '{}' extension", ignoredPaths.size(), extension);

        Set<Path> clonedSet = new HashSet<>(ignoredPaths);
        for (Path path : clonedSet) {
            if (extension.equals(getExtension(path))) { // <------------------------------------------
                checkPath(path, CREATE);
                ignoredPaths.remove(path);
            }
        }

        logger.debug("Finished processing ignored paths for '{}' extension. {} ignored paths remain", extension,
                ignoredPaths.size());
    }
```

Unfortunately, for this log, I did not have the logging on, so we would see the output of

>logger.debug("Processing {} ignored paths for '{}' extension", ignoredPaths.size(), extension);

If this would help, let me know what the right setting would be to activate the logging for exactly this.

--- a/bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/folder/FolderObserver.java
+++ b/bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/folder/FolderObserver.java
@@ -24,6 +24,7 @@ import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
@@ -113,7 +114,8 @@ public class FolderObserver implements WatchService.WatchEventListener {
         parsers.remove(extension);
 
         Set<String> removed = modelRepository.removeAllModelsOfType(extension);
-        ignoredPaths.addAll(removed.stream().map(namePathMap::get).collect(Collectors.toSet()));
+        ignoredPaths
+                .addAll(removed.stream().map(namePathMap::get).filter(Objects::nonNull).collect(Collectors.toSet()));
     }
 
     @Activate

